packages/block-library/src/archive-title/index.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Server-side rendering of the `core/archive-title` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/archive-title` block on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the filtered archive title based on the queried object.
+ */
+function render_block_core_archive_title( $attributes, $content, $block ) {
+	if ( ! is_archive() ) {
+		return '';
+	}
+
+	$tag_name           = isset( $attributes['level'] ) ? 'h' . $attributes['level'] : 'h2';
```

Comment:
`$attributes['level']` needs some escaping here as it could be anything. Otherwise `$tag_name` could pose risk for XSS. A simple `(int) $attributes['level']` might be enough.

Reply:
Thanks for spotting this :)